### ISSUE TITLE
Allow PR title editing

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#26833]
### Describe the proposed change(s).

When the PR is created, and any time in future if the PR title is changed by Renovate, store a hash of that title in the PR metadata (e.g. where we store Renovate created-in version). This way we can know if a PR title has been changed by a user or other bot, because there'll be a mismatch between the hash of the current PR title versus what we saved in the PR body.

If we decide there is a PR title mismatch (which can happen if the PR content changes, config changes, OR a user manually changes it) then we use this hash to decide if it's the last reason or not. If it's the last reason then we should note that we won't change the PR title because it's been edited.

For existing PRs without a hash, we should add it any time we update the PR body for other reasons and not proactively.

Then, we should enrich our findPr logic to find based on hash first, title second (older PRs won't have a hash). This is most important for finding closed ignored PRs when the title was manually changed before closing.

**Comment by viceice** [comment]
This also only works for platforms with hidden html comments. We should probably also store it in repo cache.

**Comment by rarkins** [comment]
Yes to repo cache. And for other platforms we should think about using the debug data in PR body even if it's non hidden

**Comment by olivier-schmitt-sonarsource** [comment]
hi,
any news on this one? 
This Renovate behavior is breaking our JIRA automation.
